### PR TITLE
spike: sign with XRPL

### PR DESCRIPTION
**Pull request by aidan-starke**
## Summary

 - Implement `signWithXrpl`

## Checklist

- [x] Add description
- [ ] Add appropriate tests

### CODE REVIEW

**Review by ken-futureverse — COMMENTED** [review]
I'll take a closer look tomorrow afternoon, but straight off the bat, is there anyway we can avoid having the entire frontend app just for testing the xrpl?

**Comment by aidan-starke** [comment]
> I'll take a closer look tomorrow afternoon, but straight off the bat, is there anyway we can avoid having the entire frontend app just for testing the xrpl?

I reckon we can just dump it - was mostly just a sanity check for me

**Comment by ken-futureverse** [comment]
Looks good, but the build is failing and what do I need to do to get e2e works on my local?

File: packages/extrinsic/package.json [hunk 1]
```diff
@@ -16,11 +16,14 @@
...
-    "neverthrow": "^6.1.0"
+    "codechain-primitives": "^1.0.4",
+    "neverthrow": "^6.1.0",
+    "xrpl": "^3.0.0",
+    "xrpl-binary-codec-prerelease": "^8.0.1"
   },
   "devDependencies": {
     "dotenv": "^16.3.2",
-    "ethers": "^6.6.5"
+    "ethers": "^5.7.2"
```
[the remark this answers is not in the document]
Comment (aidan-starke) [reply packages/extrinsic/package.json:32]: Resolved in `spike/update-api-types`

File: packages/extrinsic/src/libs/signWithXrplWallet.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+async function createSignatureOptions(
+	api: ApiPromise,
+	senderAddress: string,
+	extrinsic: Extrinsic
+) {
+	const [blockNumber, signingInfo, nonce] = await Promise.allSettled([
+		api.query.system.number(),
+		api.derive.tx.signingInfo(senderAddress),
```
Comment (KarishmaBothara) [inline packages/extrinsic/src/libs/signWithXrplWallet.ts:63]: signingInfo will give you nonce... won't need extra call to get nonce
https://github.com/polkadot-js/api/blob/master/packages/api-derive/src/tx/signingInfo.ts#L106

File: packages/extrinsic/src/libs/signWithXrplWallet.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
+import { ApiPromise } from "@polkadot/api";
+import { BN } from "@polkadot/util";
```
Comment (KarishmaBothara) [inline packages/extrinsic/src/libs/signWithXrplWallet.ts:2]: unused import can be removed

File: packages/extrinsic/src/libs/signWithXrplWallet.ts (hunk 2 above)
Comment (aidan-starke) [reply packages/extrinsic/src/libs/signWithXrplWallet.ts:62]: good point, cheers!

**Review by KarishmaBothara — APPROVED** [review]
lgtm

**Pull request merged**